src/ChainSafe.GamingSdk.Gelato/GelatoClient.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.Threading.Tasks;
+using ChainSafe.GamingSdk.Gelato.Relay;
+using ChainSafe.GamingSdk.Gelato.Types;
+using ChainSafe.GamingWeb3.Environment;
+using Nethereum.Hex.HexTypes;
+
+namespace ChainSafe.GamingSdk.Gelato
+{
+    public class GelatoClient
+    {
+        private const string GelatoRelayUrl = "https://api.gelato.digital"; // Relay GW
+        public const string SignedTypedDataV4 = "eth_signTypedData_v4";
+        private const string DefaultInternalErrorMessage = "Internal Error";
+        public const int DefaultDeadlineGap = 86_400; // 24H
+        public const string GelatoRelayErc2771Address = "0xb539068872230f20456CF38EC52EF2f91AF4AE49";
+        public const string GelatoRelay1BalanceErc2771Address = "0xd8253782c45a12053594b9deB72d8e8aB2Fca54c";
+        public const string GelatoRelayErc2771ZksyncAddress = "0x22DCC39b2AC376862183dd35A1664798dafC7Da6";
+        public const string GelatoRelay1BalanceErc2771ZksyncAddress = "0x97015cD4C3d456997DD1C40e2a18c79108FCc412";
+        public static readonly string UserNonceAbi = "[{\"inputs\": [{\"internalType\": \"address\",\"name\": \"account\",\"type\": \"address\"}],\"name\": \"userNonce\",\"outputs\": [{\"internalType\": \"uint256\",\"name\": \"\",\"type\": \"uint256\"}],\"stateMutability\": \"view\",\"type\": \"function\"}]\"}";
+        private IHttpClient httpClient;
+        private Config config;
+
+        public GelatoClient(IHttpClient httpClient, Config config)
+        {
+            this.httpClient = httpClient;
+            this.config = config;
+        }
+
+        public async Task<TResponse> Post<TRequest, TResponse>(RelayCall relayCall, TRequest request)
+        {
+            switch (relayCall)
+            {
+                case RelayCall.CallWithSyncFee:
+                    return (await this.httpClient.Post<TRequest, TResponse>($"{config.Url}/relays/v2/call-with-sync-fee", request)).Response;
```

Comment:
Use `EnsureResponse()`. `Response` may be null if there is an error.

Reply:
Will do